pkl-cli/pkl-cli.gradle.kts

```diff
@@ -237,6 +237,17 @@ fun Exec.configureExecutable(
   workingDir(outputFile.map { it.asFile.parentFile })
   executable = "${graalVm.baseDir}/bin/$nativeImageCommandName"
 
+  // For any system properties starting with `pkl.native`, strip off that prefix and pass the rest
+  // through as arguments to native-image.
+  //
+  // Allow setting args using flags like
+  // (-Dpkl.native-Dpolyglot.engine.userResourceCache=/my/cache/dir) when building through Gradle.
+  val extraArgsFromProperties =
+    System.getProperties().mapNotNull { (key, value) ->
+      if (!key.toString().startsWith("pkl.native")) null
```

Comment:
I don't like the negation here 🫣
Could we use filter with just map instead?
I guess it reads a bit better
`filter { key.startsWith(pkl.native) }`.
Or does it have to be filterNot? 🤔 😂

Reply:
Yeah; fair, switched!